### PR TITLE
InternalClusterInfoService should not ignore hidden indices

### PR DESCRIPTION
**Pull request by tlrx**
Today `InternalClusterInfoService` ignores hidden indices when retrieving shard stats of the cluster. This can lead to suboptimal shard allocation decisions as the size of shards are taken into account when allocating new shards or rebalancing existing shards.

### CODE REVIEW

**Comment by tlrx** [comment]
Note that #52423 moved ML indices to be `hidden`

**Review by original-brownbear — APPROVED** [review]
LGTM (from my end especially with the ML thing out of the way, maybe let Henning double check as well though :))

File: server/src/internalClusterTest/java/org/elasticsearch/cluster/ClusterInfoServiceIT.java [hunk 1]
```diff
@@ -117,13 +118,18 @@ private void setClusterInfoTimeout(String timeValue) {
 
     public void testClusterInfoServiceCollectsInformation() {
         internalCluster().startNodes(2);
-        assertAcked(prepareCreate("test").setSettings(Settings.builder()
-            .put(Store.INDEX_STORE_STATS_REFRESH_INTERVAL_SETTING.getKey(), 0)
-            .put(EnableAllocationDecider.INDEX_ROUTING_REBALANCE_ENABLE_SETTING.getKey(), EnableAllocationDecider.Rebalance.NONE).build()));
+
+        final String indexName = (rarely() ? "." : "") + randomAlphaOfLength(5).toLowerCase(Locale.ROOT);
```
Comment (henningandersen) [inline server/src/internalClusterTest/java/org/elasticsearch/cluster/ClusterInfoServiceIT.java:122]: nit: just use randomBoolean()?

**Review by henningandersen — APPROVED** [review]
LGTM2, thanks @tlrx .

I wonder if there is a similar issue for system indices? Or are those captured by this change?

File: server/src/internalClusterTest/java/org/elasticsearch/cluster/ClusterInfoServiceIT.java (hunk 1 above)
Comment (tlrx) [reply server/src/internalClusterTest/java/org/elasticsearch/cluster/ClusterInfoServiceIT.java]: done

**Comment by tlrx** [comment]
> I wonder if there is a similar issue for system indices? Or are those captured by this change?

@henningandersen System indices should be correctly resolved using the previous indices options, but just in case I pushed https://github.com/elastic/elasticsearch/pull/62995/commits/99b262fb03ca6bb93aff05d7493445693a413b4d so that the test is also randomly executed on a system index. Let me know what you think please

**Comment by tlrx** [comment]
@elasticmachine update branch

**Review by henningandersen — APPROVED** [review]
LGTM, thanks for also verifying that this works for system indices.

**Comment by tlrx** [comment]
@elasticmachine run elasticsearch-ci/packaging-sample-windows

**Comment by tlrx** [comment]
Thanks Armin & Henning!

**Pull request merged**